tests/bdd/features/cordon/node/test_cordon_node.py

```diff
@@ -0,0 +1,451 @@
+@pytest.fixture(scope="function")
+def create_request():
+    return {}
+
+
+###### utility functions ######
+
+
+def cordon_node(node_name, label):
+    ApiClient.nodes_api().put_node_cordon(node_name, label)
+    assert is_cordoned(node_name) == True
+
+
+def create_volume(vol_uuid):
+    request = CreateVolumeBody(
+        VolumePolicy(False), NUM_VOLUME_REPLICAS, VOLUME_SIZE, False
+    )
+    ApiClient.volumes_api().put_volume(vol_uuid, request)
+    # Check that the volume was created.
+    volumes = ApiClient.volumes_api().get_volume(vol_uuid)
+
+
+def create_volume_and_fail(vol_uuid):
+    request = CreateVolumeBody(
+        VolumePolicy(False), NUM_VOLUME_REPLICAS, VOLUME_SIZE, False
+    )
+    try:
+        ApiClient.volumes_api().put_volume(vol_uuid, request)
+        assert False
+    except Exception as e:
+        print("failed, as expected")
+    # Check that the volume wasn't created.
+    try:
+        volumes = ApiClient.volumes_api().get_volume(vol_uuid)
+        assert False
+    except Exception as e:
+        print("no volume, as expected")
+
+
+def delete_volume(vol_uuid):
+    ApiClient.volumes_api().del_volume(vol_uuid)
+
+
+def delete_all_volumes():
+    volumes = ApiClient.volumes_api().get_volumes()
+    for vol in volumes.entries:
+        ApiClient.volumes_api().del_volume(vol.spec.uuid)
+
+
+def get_volumes():
+    return ApiClient.volumes_api().get_volumes()
+
+
+def is_cordoned(node_name):
+    node = ApiClient.nodes_api().get_node(node_name)
+    present = False
+    try:
+        print("%v", node.spec.cordondrainstate)
+        assert node.spec.cordondrainstate["cordonedstate"]["cordonlabels"] != []
+        return True
+    except AttributeError as e:
+        return False
+
+
+def remove_all_cordons(node_name):
+    if is_cordoned(node_name):
+        node = ApiClient.nodes_api().get_node(node_name)
+        for label in node.spec.cordondrainstate["cordonedstate"]["cordonlabels"]:
+            ApiClient.nodes_api().delete_node_cordon(node_name, label)
+
+
+def share_volume(vol_uuid):
+    ApiClient.volumes_api().put_volume_share(vol_uuid, "nvmf")
+
+
+def uncordon_node(node_name, label):
+    try:
+        ApiClient.nodes_api().delete_node_cordon(node_name, label)
+        pytest.command_failed = False
+    except Exception as e:
+        pytest.command_failed = True
+
+
+############### scenarios ##################
+
+
+@scenario("feature.feature", "Cordoning a cordoned node")
+def test_cordoning_a_cordoned_node():
+    """Cordoning a cordoned node."""
+
+
+@scenario("feature.feature", "Cordoning a node")
+def test_cordoning_a_node():
+    """Cordoning a node."""
+
+
+# Disabled because the BDD relates to kubectl idempotent behaviour which
+# differs from REST-layer behaviour - which will fail.
+# @scenario('feature.feature', 'Cordoning a node with an existing label')
+# def test_cordoning_a_node_with_an_existing_label():
+#    """Cordoning a node with an existing label."""
```

Comment:
You can uncomment and add skip flag so it won't run

Reply:
It looks like you can't skip a scenario, only "given" "when" etc.

Reply:
Added skip directives to disable the tests.

Reply:
hmm I didn't know you could skip the verbs, you can skip a scenario:
https://github.com/openebs/mayastor-control-plane/blob/4b1086adb3c05b394af1840f44a33412173ac4f4/tests/bdd/features/csi/controller/test_controller.py#L163

Reply:
Oh right. I wasn't using the right syntax. Fixed now, thanks.